mitiq/lre/inference/multivariate_richardson.py

```
@@ -0,0 +1,128 @@
+    terms.
+
+    Args:
+        num_layers: Number of layers in the input circuit.
+        degree: Degree of the multivariate polynomial.
+
+    Returns:
+        monomial_basis: Monomial basis terms required for multivariate
+            extrapolation upto max degree
+    """
+    variable_combinations = _create_variable_combinations(num_layers, degree)
+
+    monomial_basis = []
+    for combo in variable_combinations:
+        monomial_parts = []
+        counts = Counter(combo)
+        # Ensure variables are processed in lexicographical order
+        for var in sorted(counts.keys()):
+            count = counts[var]
+            if count > 1:
+                monomial_parts.append(f"{var}**{count}")
+            else:
+                monomial_parts.append(var)
+        monomial = "*".join(monomial_parts)
+        # Handle the case where degree is 0
+        # the tuple in variable_combinations is empty
+        if not monomial:
+            monomial = "1"
+        monomial_basis.append(monomial)
+    # "1" should be the first monomial (degree = 0).
+    # max degree should be the last term
+    return monomial_basis[::-1]
+
+
+def sample_matrix(
+    input_circuit: Circuit,
+    degree: int,
+    fold_multiplier: int,
+    num_chunks: Optional[int] = None,
+):
+    """Calculates the sample matrix required for extrapolation."""
+    if degree < 1:
+        raise ValueError(
+            "Multinomial degree must be greater than or equal to 1."
+        )
+    if fold_multiplier < 1:
+        raise ValueError("Fold multiplier must be greater than or equal to 1.")
+
+    num_layers = _get_num_layers_without_measurements(input_circuit)
+
+    scale_factor_vectors = _get_scale_factor_vectors(
+        input_circuit, degree, fold_multiplier, num_chunks
+    )
+
+    monomial_terms = full_monomial_basis(num_layers, degree)
+    if len(monomial_terms) != len(scale_factor_vectors):
+        raise ValueError("Sample matrix will not be a square matrix.")
```

Check if we need to raise this error.